### PR TITLE
feat (buttons): adds button functionality (Dev-691)

### PR DESCRIPTION
**Pull request by domsteinbach**
resolves DEV-691

### CODE REVIEW

File: src/Buttons/Button.elm [hunk 1]
```diff
@@ -1,46 +1,81 @@
...
--- convenience function: returns a basicButton with the Variant "Primary"
-primaryButton : List (Attribute msg) -> String -> BasicButtonSize -> Node msg
+primaryButton : List (Attribute msg) -> String -> BasicButtonSize -> Html.Styled.Html msg
 primaryButton attrs txt size =
     basicButton attrs txt size Primary
 
--- convenience function: returns a basicButton with the Variant "Secondary"
-secondaryButton : List (Attribute msg) -> String -> BasicButtonSize -> Node msg
+
+primaryButtonUnStyled : List (Attribute msg) -> String -> BasicButtonSize -> Html msg
```
Comment (mdelez) [inline src/Buttons/Button.elm:18]: UnStyled -> Unstyled

File: src/Buttons/Button.elm [hunk 2]
```diff
@@ -1,46 +1,81 @@
...
+
+
+secondaryButton : List (Attribute msg) -> String -> BasicButtonSize -> Html.Styled.Html msg
 secondaryButton attrs txt size =
     basicButton attrs txt size Secondary
 
--- convenience function: returns a basicButton with the Variant "White"
-whiteButton : List (Attribute msg) -> String -> BasicButtonSize -> Node msg
+
+secondaryButtonUnStyled : List (Attribute msg) -> String -> BasicButtonSize -> Html msg
```
Comment (mdelez) [inline src/Buttons/Button.elm:29]: UnStyled -> Unstyled

**Review by mdelez — CHANGES_REQUESTED** [review]
I gotta be honest, I really don't like this 'ToUnstyled' version of each button. Perhaps we only need this temporarily but can't we just change everything to use `Html.Styled` and then each button will have `\> toUnstyled` at the end?

File: src/Buttons/Button.elm (hunk 1 above)
Comment (domsteinbach) [reply src/Buttons/Button.elm:18]: I do not like it either. Having it temporarily was the idea. There is need to change the all the modules/components which use Html to Htlm.Styled as well as all other modules which return the toUnstyled version. This is only a step on the path.

**Comment by domsteinbach** [comment]
Closed due to a lot of refactoring by using own css classes.

**Pull request closed without merging**